integration/update_field_compat_test.go

```diff
@@ -481,6 +481,80 @@ func TestUpdateFieldCompatMin(t *testing.T) {
 	testUpdateCompat(t, testCases)
 }
 
+func TestUpdateFieldCompatRename(t *testing.T) {
+	testCases := map[string]updateCompatTestCase{
+		"Simple": {
+			update: bson.D{{"$rename", bson.D{{"v", "foo"}}}},
+		},
+		"DuplicateField": {
+			update:     bson.D{{"$rename", bson.D{{"v", "v"}}}},
+			resultType: emptyResult,
+		},
+		"NonExistingField": {
+			update:     bson.D{{"$rename", bson.D{{"foo", "bar"}}}},
+			resultType: emptyResult,
+		},
+		"EmptyField": {
+			update:     bson.D{{"$rename", bson.D{{"", "v"}}}},
+			resultType: emptyResult,
+		},
+		"EmptyDest": {
+			update:     bson.D{{"$rename", bson.D{{"v", ""}}}},
+			resultType: emptyResult,
+		},
+		"DotDocumentMove": {
+			update:        bson.D{{"$rename", bson.D{{"v.foo", "boo"}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1776",
+		},
+		"DotDocumentDuplicate": {
+			update:        bson.D{{"$rename", bson.D{{"v.foo", "v.array"}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1776",
+		},
+		"DotDocumentNonExisting": {
+			update:     bson.D{{"$rename", bson.D{{"foo.bar", ""}}}},
+			resultType: emptyResult,
+		},
+		"DotArrayField": {
+			update:     bson.D{{"$rename", bson.D{{"v.array.0", ""}}}},
+			resultType: emptyResult,
+		},
+		"DotArrayNonExisting": {
+			update:     bson.D{{"$rename", bson.D{{"foo.0.baz", int32(1)}}}},
+			resultType: emptyResult,
+		},
+		"Multiple": {
+			update:        bson.D{{"$rename", bson.D{{"v.foo", "v.bar"}, {"v.42", "v.43"}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1776",
```

Comment:
👍 Thanks for creating a separate issue! I think after my PR [about array checks](https://github.com/FerretDB/FerretDB/pull/1778), I have an idea of how to add some data to cover such test case. I like the idea of having a separate issue, so when this PR and my PR #1778 are merged, we can quickly add test cases in #1776 ;)